### PR TITLE
Fix to support server resend

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: app/src/main/java/io/rapidpro/androidchannel/payload/MTBroadcast.java [hunk 1]
```diff
@@ -77,6 +77,12 @@ public void execute(Context context, SyncPayload sent){
                     RapidPro.LOG.d("Resending DELIVERED notification for " + msg.getPhone() + " id: " + to.getId());
                     DBCommandHelper.queueCommand(context, new MTTextDelivered(to.getId(), msg.getPhone()));
                 }
+
+                // Resend from server
+                if (state == MTTextMessage.FAILED || state == MTTextMessage.FAILED_SYNCED) {
+                    RapidPro.LOG.d("Server request to resend for " + msg.getPhone() + " id: " + to.getId());
+                    DBCommandHelper.updateCommandStateWithServerId(context, MTTextMessage.CMD, to.getId(), MTTextMessage.RETRY, "" + 0);
+                }
```
Comment (norkans7) [inline app/src/main/java/io/rapidpro/androidchannel/payload/MTBroadcast.java:85]: We use to rely on the fact the resent messages used new ids, now that we use the same IDs, we need to update and set the retries back to 0 on the same command

**Review by nicpottier — APPROVED** [review]
Nice, tested?

**Review by rowanseymour — APPROVED** [review]

**Comment by norkans7** [comment]
Not sure how to make the message fail in the same app the same way, going through the code that seem to be the issue now.
Also the bug would not be a problem if the resend happened after we trimmed the commands
https://github.com/rapidpro/android-channel/blob/16c3618ffdbc6351c15447e32f4d84bb0be6c14f/app/src/main/java/io/rapidpro/androidchannel/data/DBCommandHelper.java#L133

**Comment by nicpottier** [comment]
You could send to an invalid number ya? +2509999999

**Comment by norkans7** [comment]
This seems to not work at all.
I'll check on that again

**Pull request closed without merging**